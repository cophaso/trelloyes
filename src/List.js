import React from 'react';
import Card from './Card.js';


function List(props){
  return(
    <section className="List" >
      <header className="List-header">
        <h2>{props.header}</h2>
        <button type="button"
        onClick={() => props.onClickAdd(props.id)}>Add New Card</button>
      </header>
      { props.cardIds.map ( (card, i) => {
        return(
          <Card 
          title={card.title}
          content={card.content}
          key={i}
          id={card.id}
          onClickDelete={props.onClickDelete}></Card>
        )
      })
      }
    </section>
  )
}

List.defaultProps = {
  onClickAdd: () => {},
}

export default List;