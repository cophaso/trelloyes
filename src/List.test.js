import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';

describe('List Test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header="Test Header" cardIds={['a', 'b']} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List header="Test Header" cardIds={['a', 'b']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});