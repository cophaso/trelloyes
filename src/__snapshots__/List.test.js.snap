// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Test renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test Header
    </h2>
    <button
      onClick={[Function]}
      type="button"
    >
      Add New Card
    </button>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
  </div>
</section>
`;
